Realm/RLMAccessor.mm

```diff
@@ -467,24 +518,33 @@ id unmanagedGetter(RLMProperty *prop, const char *) {
 }
 
 id unmanagedSetter(RLMProperty *prop, const char *) {
-    // Only RLMArray needs special handling for the unmanaged setter
-    if (!prop.array) {
+    // Only RLMArray & RLMSet needs special handling for the unmanaged setter
+    if (!prop.collection) {
         return nil;
     }
 
     NSString *propName = prop.name;
     return ^(RLMObjectBase *obj, id<NSFastEnumeration> values) {
         auto prop = obj->_objectSchema[propName];
         RLMValidateValueForProperty(values, obj->_objectSchema, prop, true);
-
-        // make copy when setting (as is the case for all other variants)
-        RLMArray *ar;
-        if (prop.type == RLMPropertyTypeObject)
-            ar = [[RLMArray alloc] initWithObjectClassName:prop.objectClassName];
-        else
-            ar = [[RLMArray alloc] initWithObjectType:prop.type optional:prop.optional];
-        [ar addObjects:values];
-        superSet(obj, propName, ar);
+        id collection;
+        if (prop.array) {
```

Comment:
IMO that would hurt the readability. Unless something like this was done but it probably is another little thing that will increase compile time.

```
id collection = [prop](Class cls, bool object) {
            return object ? [[cls alloc] initWithObjectClassName:prop.objectClassName] :
                            [[cls alloc] initWithObjectType:prop.type optional:prop.optional];
        };
```

Reply:
```objc
        Class cls = prop.array ? RLMArray.class : RLMSet.class;
        // make copy when setting (as is the case for all other variants)
        if (prop.type == RLMPropertyTypeObject)
            collection = [[cls alloc] initWithObjectClassName:prop.objectClassName];
        else
            collection = [[cls alloc] initWithObjectType:prop.type optional:prop.optional];
```
If you still think it's not readable (I think it's fine), I'd consider that, since there is so much repeated code between the types, that maybe RLMCollection should own some of that.

Reply:
I would consider doing this for a number of the introduced conditionals. There are many `if (prop.array)`s with identical code between them, bar the instantiating type.